### ISSUE TITLE
Punk Girl transforming to default female hero

### ISSUE THREAD

**Issue by Lexx2k** [BGforgeNet/Fallout2_Hero_Appearance#1]
Just noticed an issue with the mod in Fallout et Tu:

Playing as punk girl and getting knocked down / dying will play the regular female animations, even though all the necessary `*b*.frm` files are existing.

I've tried the same with the files from this repo + a new game in Fo2 with RP, with exactly the same result.

I have no idea what exactly is causing this.

**Comment by Lexx2k** [comment]
I did some more tests: The same issue happens with other female appearances, but NOT with the males. Bald dude remains bald dude on death, etc. 

/Edit: Just realized this mod is integrated in Sfall. Guess I should have opened the ticket over there instead. Oh well.

**Comment by Lexx2k** [comment]
Alright, the issue is that all `*b*.frm` animations need to be `*.fr0, 1, 2,..` splits. With that, everything works as it should.

**Comment by burner1024** [comment]
Cool, thanks. How do you split them, though? Need to do the same for other girls.

**Comment by Lexx2k** [comment]
I used the FRM Animator to first batch convert all `*b*.frm` files to *.fpr (project files). After that I batch converted all project files to `*.fr*`. It's not a too big deal. Most annoying is that you have to filter out all the b-animations first.

**Comment by burner1024** [comment]
I'm seeing some male anims on the list, presumably they also need to be converted?
Also, I assume you mean Frame Animator? Which version did you use? I see [2.7](https://www.nma-fallout.com/threads/solved-frame-animator-2-7-issues.212615/) mentioned, but the latest I can find is [2.5](https://teamx.bgforge.net/utils/index.html). 
```
./hfr00s01/art/critters/_hfjmpsba.frm
./hfr00s01/art/critters/_hfjmpsbb.frm
./hfr00s01/art/critters/_hfjmpsbd.frm
./hfr00s01/art/critters/_hfjmpsbe.frm
./hfr00s01/art/critters/_hfjmpsbf.frm
./hfr00s01/art/critters/_hfjmpsbg.frm
./hfr00s01/art/critters/_hfjmpsbh.frm
./hfr00s01/art/critters/_hfjmpsbi.frm
./hfr00s01/art/critters/_hfjmpsbj.frm
./hfr00s01/art/critters/_hfjmpsbk.frm
./hfr00s01/art/critters/_hfjmpsbl.frm
./hfr00s01/art/critters/_hfjmpsbm.frm
./hfr00s01/art/critters/_hfjmpsbn.frm
./hfr00s01/art/critters/_hfjmpsbo.frm
./hfr00s01/art/critters/_hfjmpsbp.frm
./hfr00s01/art/critters/_hflthrba.frm
./hfr00s01/art/critters/_hflthrbb.frm
./hfr00s01/art/critters/_hflthrbd.frm
./hfr00s01/art/critters/_hflthrbe.frm
./hfr00s01/art/critters/_hflthrbf.frm
./hfr00s01/art/critters/_hflthrbg.frm
./hfr00s01/art/critters/_hflthrbh.frm
./hfr00s01/art/critters/_hflthrbi.frm
./hfr00s01/art/critters/_hflthrbj.frm
./hfr00s01/art/critters/_hflthrbk.frm
./hfr00s01/art/critters/_hflthrbl.frm
./hfr00s01/art/critters/_hflthrbm.frm
./hfr00s01/art/critters/_hflthrbn.frm
./hfr00s01/art/critters/_hflthrbo.frm
./hfr00s01/art/critters/_hflthrbp.frm
./hfr00s01/art/critters/_hfmaxxba.frm
./hfr00s01/art/critters/_hfmaxxbb.frm
./hfr00s01/art/critters/_hfmaxxbd.frm
./hfr00s01/art/critters/_hfmaxxbe.frm
./hfr00s01/art/critters/_hfmaxxbf.frm
./hfr00s01/art/critters/_hfmaxxbg.frm
./hfr00s01/art/critters/_hfmaxxbh.frm
./hfr00s01/art/critters/_hfmaxxbi.frm
./hfr00s01/art/critters/_hfmaxxbj.frm
./hfr00s01/art/critters/_hfmaxxbk.frm
./hfr00s01/art/critters/_hfmaxxbl.frm
./hfr00s01/art/critters/_hfmaxxbm.frm
./hfr00s01/art/critters/_hfmaxxbn.frm
./hfr00s01/art/critters/_hfmaxxbo.frm
./hfr00s01/art/critters/_hfmaxxbp.frm
./hfr00s01/art/critters/_hfmetlba.frm
./hfr00s01/art/critters/_hfmetlbb.frm
./hfr00s01/art/critters/_hfmetlbd.frm
./hfr00s01/art/critters/_hfmetlbe.frm
./hfr00s01/art/critters/_hfmetlbf.frm
./hfr00s01/art/critters/_hfmetlbg.frm
./hfr00s01/art/critters/_hfmetlbh.frm
./hfr00s01/art/critters/_hfmetlbi.frm
./hfr00s01/art/critters/_hfmetlbj.frm
./hfr00s01/art/critters/_hfmetlbk.frm
./hfr00s01/art/critters/_hfmetlbl.frm
./hfr00s01/art/critters/_hfmetlbm.frm
./hfr00s01/art/critters/_hfmetlbn.frm
./hfr00s01/art/critters/_hfmetlbo.frm
./hfr00s01/art/critters/_hfmetlbp.frm
./hfr00s01/art/critters/_hfprimba.frm
./hfr00s01/art/critters/_hfprimbb.frm
./hfr00s01/art/critters/_hfprimbd.frm
./hfr00s01/art/critters/_hfprimbe.frm
./hfr00s01/art/critters/_hfprimbf.frm
./hfr00s01/art/critters/_hfprimbg.frm
./hfr00s01/art/critters/_hfprimbh.frm
./hfr00s01/art/critters/_hfprimbi.frm
./hfr00s01/art/critters/_hfprimbj.frm
./hfr00s01/art/critters/_hfprimbk.frm
./hfr00s01/art/critters/_hfprimbl.frm
./hfr00s01/art/critters/_hfprimbm.frm
./hfr00s01/art/critters/_hfprimbn.frm
./hfr00s01/art/critters/_hfprimbo.frm
./hfr00s01/art/critters/_hfprimbp.frm
./hfr00s02/art/critters/_hfjmpsba.frm
./hfr00s02/art/critters/_hfjmpsbb.frm
./hfr00s02/art/critters/_hfjmpsbd.frm
./hfr00s02/art/critters/_hfjmpsbe.frm
./hfr00s02/art/critters/_hfjmpsbf.frm
./hfr00s02/art/critters/_hfjmpsbg.frm
./hfr00s02/art/critters/_hfjmpsbh.frm
./hfr00s02/art/critters/_hfjmpsbi.frm
./hfr00s02/art/critters/_hfjmpsbj.frm
./hfr00s02/art/critters/_hfjmpsbk.frm
./hfr00s02/art/critters/_hfjmpsbl.frm
./hfr00s02/art/critters/_hfjmpsbm.frm
./hfr00s02/art/critters/_hfjmpsbn.frm
./hfr00s02/art/critters/_hfjmpsbo.frm
./hfr00s02/art/critters/_hfjmpsbp.frm
./hfr00s02/art/critters/_hflthrba.frm
./hfr00s02/art/critters/_hflthrbb.frm
./hfr00s02/art/critters/_hflthrbd.frm
./hfr00s02/art/critters/_hflthrbe.frm
./hfr00s02/art/critters/_hflthrbf.frm
./hfr00s02/art/critters/_hflthrbg.frm
./hfr00s02/art/critters/_hflthrbh.frm
./hfr00s02/art/critters/_hflthrbi.frm
./hfr00s02/art/critters/_hflthrbj.frm
./hfr00s02/art/critters/_hflthrbk.frm
./hfr00s02/art/critters/_hflthrbl.frm
./hfr00s02/art/critters/_hflthrbm.frm
./hfr00s02/art/critters/_hflthrbn.frm
./hfr00s02/art/critters/_hflthrbo.frm
./hfr00s02/art/critters/_hflthrbp.frm
./hfr00s02/art/critters/_hfmaxxba.frm
./hfr00s02/art/critters/_hfmaxxbb.frm
./hfr00s02/art/critters/_hfmaxxbd.frm
./hfr00s02/art/critters/_hfmaxxbe.frm
./hfr00s02/art/critters/_hfmaxxbf.frm
./hfr00s02/art/critters/_hfmaxxbg.frm
./hfr00s02/art/critters/_hfmaxxbh.frm
./hfr00s02/art/critters/_hfmaxxbi.frm
./hfr00s02/art/critters/_hfmaxxbj.frm
./hfr00s02/art/critters/_hfmaxxbk.frm
./hfr00s02/art/critters/_hfmaxxbl.frm
./hfr00s02/art/critters/_hfmaxxbm.frm
./hfr00s02/art/critters/_hfmaxxbn.frm
./hfr00s02/art/critters/_hfmaxxbo.frm
./hfr00s02/art/critters/_hfmaxxbp.frm
./hfr00s02/art/critters/_hfmetlba.frm
./hfr00s02/art/critters/_hfmetlbb.frm
./hfr00s02/art/critters/_hfmetlbd.frm
./hfr00s02/art/critters/_hfmetlbe.frm
./hfr00s02/art/critters/_hfmetlbf.frm
./hfr00s02/art/critters/_hfmetlbg.frm
./hfr00s02/art/critters/_hfmetlbh.frm
./hfr00s02/art/critters/_hfmetlbi.frm
./hfr00s02/art/critters/_hfmetlbj.frm
./hfr00s02/art/critters/_hfmetlbk.frm
./hfr00s02/art/critters/_hfmetlbl.frm
./hfr00s02/art/critters/_hfmetlbm.frm
./hfr00s02/art/critters/_hfmetlbn.frm
./hfr00s02/art/critters/_hfmetlbo.frm
./hfr00s02/art/critters/_hfmetlbp.frm
./hfr00s02/art/critters/_hfprimba.frm
./hfr00s02/art/critters/_hfprimbb.frm
./hfr00s02/art/critters/_hfprimbd.frm
./hfr00s02/art/critters/_hfprimbe.frm
./hfr00s02/art/critters/_hfprimbf.frm
./hfr00s02/art/critters/_hfprimbg.frm
./hfr00s02/art/critters/_hfprimbh.frm
./hfr00s02/art/critters/_hfprimbi.frm
./hfr00s02/art/critters/_hfprimbj.frm
./hfr00s02/art/critters/_hfprimbk.frm
./hfr00s02/art/critters/_hfprimbl.frm
./hfr00s02/art/critters/_hfprimbm.frm
./hfr00s02/art/critters/_hfprimbn.frm
./hfr00s02/art/critters/_hfprimbo.frm
./hfr00s02/art/critters/_hfprimbp.frm
./hmr00s01/art/critters/_hmjmpsbn.frm
./hmr00s01/art/critters/_hmwarrba.frm
./hmr00s01/art/critters/_hmwarrbb.frm
./hmr00s01/art/critters/_hmwarrbd.frm
./hmr00s01/art/critters/_hmwarrbf.frm
./hmr00s01/art/critters/_hmwarrbg.frm
./hmr00s01/art/critters/_hmwarrbi.frm
./hmr00s01/art/critters/_hmwarrbl.frm
./hmr00s01/art/critters/_hmwarrbm.frm
./hmr00s01/art/critters/_hmwarrbo.frm
./hmr00s01/art/critters/_hmwarrbp.frm
./hmr00s02/art/critters/_hmjmpsbn.frm
./hmr00s02/art/critters/_hmwarrba.frm
./hmr00s02/art/critters/_hmwarrbb.frm
./hmr00s02/art/critters/_hmwarrbd.frm
./hmr00s02/art/critters/_hmwarrbf.frm
./hmr00s02/art/critters/_hmwarrbg.frm
./hmr00s02/art/critters/_hmwarrbi.frm
./hmr00s02/art/critters/_hmwarrbl.frm
./hmr00s02/art/critters/_hmwarrbm.frm
./hmr00s02/art/critters/_hmwarrbo.frm
./hmr00s02/art/critters/_hmwarrbp.frm
```

**Comment by Lexx2k** [comment]
Don't know about males. I haven't seen any issues with bald guy (which I am always using to play).
I'm using Frame Animator 2.70 apparently. Don't ask me from where I got it... I'm using these files since at least 2006.

**Comment by burner1024** [comment]
> Don't know about males. I haven't seen any issues with bald guy (which I am always using to play).

These are warr/jmp, so you'd probably not see them if you hadn't died until Klamath.

Just to confirm - @FakelsHub, @NovaRain is this normal behaviour (frms only working when split)?

**Comment by NovaRain** [comment]
Yes, [it's been discussed on NMA before](https://www.nma-fallout.com/threads/fallout-frm-archive.216917/page-2#post-4313048). You need to replace all files in original hero set 1:1, in exact file names.

**Comment by burner1024** [comment]
OK then. Maybe add a note to the config?

**Comment by burner1024** [comment]
I re-checked everything vs original files list:
1. It appears that death anims are good in both male suites, but a few others are missing. (I remember there was a list of animation codes somethere in the wiki, but can't find it now, so not sure what are those).
2. For females, `*primb[a/b/o/p].frm` aren't supposed to be split. I reverted `a/b` for Punk Girl.
3. Still, Punk Girl is missing more stuff.
```diff

diff -u male_frm.list ms01.list  | grep '^-'
--- male_frm.list	2020-02-29 19:37:48.593027328 +0000
-hmjmpsgg.frm
-hmjmpsna.frm
-hmlthrau.frm
-hmmetlau.frm

diff -u male_frm.list ms02.list  | grep '^-'
--- male_frm.list	2020-02-29 19:37:48.593027328 +0000
-hmjmpsgg.frm
-hmjmpsna.frm
-hmlthrau.frm
-hmmetlau.frm

diff -u female_frm.list fs03.list | grep '^-'
--- female_frm.list	2020-02-29 19:37:48.593027328 +0000
-hfjmpsgg.frm
-hfjmpsrd.frm
-hfjmpsrf.frm
-hfjmpsrg.frm
-hfjmpsri.frm
-hfprimbo.frm
-hfprimbp.frm
-hfprimch.frm
-hfprimcj.frm
-hfprimra.frm
-hfprimrb.frm
```

**Comment by burner1024** [comment]
Hopefully someone comes along and adds the missing frames at some point.

**Comment by Lexx2k** [comment]
Hmm are you sure they aren't in Foxx' version?
https://www.mediafire.com/file/e6yp3tousdflj9p/PunkGirl_set_dat.rar/file
https://www.nma-fallout.com/threads/ms-punk-girl-a-new-playable-character.197710/page-15#post-4259787

I've added these to my package a while ago.

**Comment by burner1024** [comment]
I found almost everything there, yes, thanks. Only `hfjmpsgg.frm` still missing.
I got the files from 1@2 originally, I didn't know there was any other version.
`hfprimb[a/b/o/p]` are still wrong in 1@2 repo, but I guess they aren't needed there in the first place.

All this is why I'd prefer a unified mod version bundled with various distros as needed, rather than trying to get multiple copies in sync manually.
